internal/observability/events/options.go

```diff
@@ -27,6 +27,10 @@ type options struct {
 	withRequest     *Request
 	withResponse    *Response
 	withAuth        *Auth
+
+	withBroker          broker // test only option
```

Comment:
minor nit: maybe `withTestBroker`

Reply:
umm... it actually is anything that implements the `broker` interface.